src/components/FileTree/index.tsx

```diff
@@ -178,15 +178,15 @@ const mapStateToProps = (
   state: ApplicationState,
   ownProps: PublicProps & DefaultProps,
```

Comment:
This seems more realistic and it would have produced a TS error:

```diff
diff --git a/src/components/FileTree/index.tsx b/src/components/FileTree/index.tsx
index d394eaa..9ceb8ad 100644
--- a/src/components/FileTree/index.tsx
+++ b/src/components/FileTree/index.tsx
@@ -176,7 +176,7 @@ export class FileTreeBase extends React.Component<Props> {
 
 const mapStateToProps = (
   state: ApplicationState,
-  ownProps: PublicProps & DefaultProps,
+  ownProps: PublicProps & Partial<DefaultProps>,
 ): PropsFromState => {
   const { versionId } = ownProps;
   const version = getVersionInfo(state.versions, versionId);
```

Reply:
I checked all other components and I don't see any incorrect types for `ownProps`. Typically we set it to `PublicProps` to be safe.

Reply:
ah right, but here it should be `ownProps: PublicProps` only as well.